js-exercises/drop-elements/dropElements.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * @template T
+ * @param {T[]} elements
+ * @param {(T) => boolean} predicate
+ * @return {T[]}
+ */
+function dropElements(elements, predicate) {
+  const firstTrueIndex = elements.findIndex(predicate);
```

Comment:
@debajyoti-xyz seems an incorrect solution!
`predicate` is not a primitive value like number or string, its a function

Reply:
@satyambnsal I am unable to understand the 2nd line.
As, I am treating `predicate` as a function both in [_JSDoc_](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#template) `(T) => boolean` and in its usage `elements.findIndex(predicate)`.

About the solution,  the README reads
> Drop the elements of an array ... until the predicate ... returns true.

But the test case suggest to *drop all elements for which predicate returns false*, similar to *Array.filter*
So I have comment out two `expect` from the given test case.

```javascript
     expect(dropElements([0, 1, 0, 1], (n) => {
       return n === 1;
     })).toEqual([1, 1]);

     expect(dropElements([1, 2, 3, 9, 2], (n) => {
       return n > 2;
     })).toEqual([3, 9]);
```